tests/valid_cases/transaction.rs

```diff
@@ -111,755 +148,742 @@ fn maturity() {
     )
     .validate(block_height)
     .err()
-    .unwrap();
+    .expect("Expected erroneous transaction");
+
     assert_eq!(ValidationError::TransactionMaturity, err);
 }
 
 #[test]
 fn max_iow() {
-    let mut rng_base = StdRng::seed_from_u64(8586);
-    let rng = &mut rng_base;
+    let rng = &mut StdRng::seed_from_u64(8586);
 
     let maturity = 100;
     let block_height = 1000;
 
-    Transaction::script(
-        rng.next_u64(),
-        MAX_GAS_PER_TX,
-        rng.next_u64(),
-        maturity,
-        rng.gen::<Witness>().into_inner(),
-        rng.gen::<Witness>().into_inner(),
-        vec![
-            Input::coin(
-                rng.gen(),
-                rng.gen(),
-                rng.next_u64(),
-                rng.gen(),
-                0,
-                rng.next_u64(),
-                vec![],
-                vec![]
-            );
-            MAX_INPUTS as usize
-        ],
-        vec![Output::coin(rng.gen(), rng.next_u64(), rng.gen()); MAX_OUTPUTS as usize],
-        vec![rng.gen(); MAX_WITNESSES as usize],
-    )
-    .validate(block_height)
-    .unwrap();
+    let secret = SecretKey::random(rng);
 
-    Transaction::create(
-        rng.next_u64(),
-        MAX_GAS_PER_TX,
-        rng.next_u64(),
-        maturity,
-        0,
-        rng.gen(),
-        vec![],
-        vec![],
-        vec![
-            Input::coin(
-                rng.gen(),
-                rng.gen(),
-                rng.next_u64(),
-                rng.gen(),
-                0,
-                rng.next_u64(),
-                vec![],
-                vec![]
-            );
-            MAX_INPUTS as usize
-        ],
-        vec![Output::coin(rng.gen(), rng.next_u64(), rng.gen()); MAX_OUTPUTS as usize],
-        vec![rng.gen(); MAX_WITNESSES as usize],
-    )
-    .validate(block_height)
-    .unwrap();
+    let mut builder = TransactionBuilder::script(generate_bytes(rng), generate_bytes(rng));
+
+    builder
+        .gas_price(rng.gen())
+        .gas_limit(MAX_GAS_PER_TX)
+        .maturity(maturity)
+        .add_unsigned_coin_input(
+            rng.gen(),
+            &secret,
+            rng.gen(),
+            rng.gen(),
+            maturity,
+            generate_bytes(rng),
+            generate_bytes(rng),
+        );
+
+    while builder.outputs().len() < MAX_OUTPUTS as usize {
+        builder.add_output(Output::coin(rng.gen(), rng.gen(), rng.gen()));
+    }
+
+    while builder.witnesses().len() < MAX_WITNESSES as usize {
+        builder.add_witness(generate_bytes(rng).into());
+    }
+
+    builder
+        .finalize()
+        .validate(block_height)
+        .expect("Failed to validate transaction");
+
+    let mut builder = TransactionBuilder::create(MAX_WITNESSES - 1, rng.gen(), vec![], vec![]);
```

Comment:
For readability we should have these as separate tests or at least a comment separator between tests explaining what is about to be tested.

Reply:
Added comments in: https://github.com/FuelLabs/fuel-tx/pull/57/commits/b0f2e785627fc60cedd531ddd985a7a79e835ba9

Also related: https://github.com/FuelLabs/fuel-tx/issues/83